src/analysis/Environments/Environments.ts

```diff
@@ -207,6 +210,10 @@ export function PauseButton(props: PauseButtonProps): ReactNode {
 type LeoAppProviderNeeds = Pick<LeoAppProvider, 'listWithoutProject' | 'get' | 'pause' | 'delete'>;
 type LeoRuntimeProviderNeeds = Pick<LeoRuntimeProvider, 'list' | 'stop' | 'delete'>;
 type LeoDiskProviderNeeds = Pick<LeoDiskProvider, 'list' | 'delete'>;
+export interface EnvironmentPermissionsProvider {
```

Comment:
Can we call this ResourcePermissionProvider? That's more in line with what we call functions that can handle `(resource: PersistenDisk | Runtime | App)` over some common field like `auditInfo`

Reply:
yeah... I think I like that.... gonna think about the Resource arg tho... might be better to define a narrow interface IHavePermissions.... which all three implement.